TelepresenceBot/server-unit/core/mover.js

```diff
@@ -0,0 +1,23 @@
+function Mover(clientsAndRooms, emitter) {
+    var hasNoRooms = function(clientId) {
+        return !clientsAndRooms[clientId];
+    };
+
+    return {
+        moveIn: function(clientId, direction) {
+            if(hasNoRooms(clientId)) {
+                return;
+            }
+
+            var roomsClientIsIn = Object.keys(clientsAndRooms[clientId]);
+
+            for(var i = 0; i < roomsClientIsIn.length; i++) {
+                emitter.to(roomsClientIsIn[i]).emit('direction', direction);
```

Comment:
this looks a little confusing, the room ids are keys in `clientsAndRooms[clientId]?


Also because we're in `JS` land we can start dropping `fori` loops in favour of `.map/.forEach`